x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/shared/map_processor_type_to_form.tsx

```diff
@@ -192,46 +281,79 @@ export const mapProcessorTypeToDescriptor: MapProcessorTypeToDescriptor = {
         />
       );
     },
+    getDefaultDescription: ({ field, policy_name: policyName, target_field: targetField }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.enrich', {
+        defaultMessage:
+          'Enriches data to "{target_field}" if the "{policy_name}" policy matches "{field}"',
+        values: {
+          field,
+          policy_name: policyName,
+          target_field: targetField,
+        },
+      }),
   },
   fail: {
     FieldsComponent: Fail,
     docLinkPath: '/fail-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.fail', {
       defaultMessage: 'Fail',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.fail', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.fail', {
       defaultMessage:
         'Returns a custom error message on failure. Often used to notify requesters of required conditions.',
     }),
+    getDefaultDescription: () =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.fail', {
+        defaultMessage: 'Raises an exception that halts execution',
+      }),
   },
   foreach: {
     FieldsComponent: Foreach,
     docLinkPath: '/foreach-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.foreach', {
       defaultMessage: 'Foreach',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.foreach', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.foreach', {
       defaultMessage: 'Applies an ingest processor to each value in an array.',
     }),
+    getDefaultDescription: ({ field, processor }) => {
+      const processorName = Object.keys(processor ?? {})[0];
+      return processorName
+        ? i18n.translate('xpack.ingestPipelines.processors.defaultDescription.foreach', {
+            defaultMessage: 'Runs the "{processorName}" processor for each object in "{field}"',
```

Comment:
```suggestion
            defaultMessage: 'Runs a processor for each object in "{field}"',
```

Reply:
Suggest removing `processorName` so that the default description displays if the user doesn't supply a processor name.